versioned_hdf5/replay.py

```diff
@@ -1,9 +1,19 @@
-from h5py import VirtualLayout, h5s, Dataset, Group, __version__ as h5py_version
+from typing import List, Iterable, Union, Dict
```

Comment:
Probably doesn't matter for what you've added, but let's use `from __future__ import annotations` and `if TYPE_CHECKING` to prevent type annotations from having any runtime effects.

Reply:
I'll add `from __future__ import annotations`, but I'm not sure how to make use of `if TYPE_CHECKING`. I'm not actually using any types defined in `versioned_hdf5` for static type checks except `VersionedHDF5File`, which is required to be defined before this module is executed anyway because it's used for some runtime type checks elsewhere in this file. If there's a better way to handle this though I'd be grateful to learn.

Reply:
Oh I guess you're right that `if TYPE_CHECKING` is only useful if you have types other than from `typing`.